xhr/resources/inspect-headers.py

```diff
@@ -1,33 +1,50 @@
+from six import PY3
+
 def get_response(raw_headers, filter_value, filter_name):
```

Comment:
It's unfortunate that this function uses `raw_headers`. It'd be great if we could refactor this to avoid using `raw_headers`. It's not obvious to me why it'd need to do that. Maybe tag someone familiar with xhr to confirm.

Reply:
@annevk might want to comment here

Reply:
How would you do the kind of checks it does without raw headers?

Reply:
raw_headers is important and I hope we're not refactoring it away in places where it has been used, as we'll lose coverage.

Reply:
It won't go away, but as you can see, this kind of encoding juggling will be needed wherever it's used.

IIUC, this function only needs to filter by header values or names. The former could be implemented by going through `request.headers.itervalues()` while the latter with `request.headers.get_list(filter_name)`.

Reply:
No, this function cares about the exact way headers are serialized. E.g., that there's a space after the colon. That they end with CRLF. It also cares about the header name not changing case. I very much doubt the higher level libraries enforce that as it's not something H/1 really enforces either way.

Reply:
I see. If the test needs to explicitly assert the syntax, then yes we'd have to do this. The parsing logic in stdlib is much laxer.

Reply:
That's what it does by doing its own parsing, right? I think this caught an issue browsers wanted to align on some years ago, but it could have been another usage of raw_headers as well.